tests/DefaultUnleashTest.php

```diff
@@ -813,6 +826,182 @@ public function testAutoRegistrationOnInvalidResponse()
         );
     }
 
+    public function testDependenciesUnresolvedFeatureVariants()
+    {
+        $data = [
+            'features' => [
+                [
+                    'name' => 'child',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'dependencies' => [
+                        [
+                            'feature' => 'parent',
+                            'variants' => ['variant1'],
+                        ],
+                    ],
+                ],
+                [
+                    'name' => 'parent',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'variants' => [
+                        [
+                            'name' => 'variant1',
+                            'weight' => 1,
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $this->pushResponse($data);
+        $instance = $this->getCachedInstance();
+        self::assertTrue($instance->isEnabled('child'));
+    }
+
+    public function testDependenciesDirectFeatureVariants()
+    {
+        $data = [
+            'features' => [
+                [
+                    'name' => 'parent',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'variants' => [
+                        [
+                            'name' => 'variant1',
+                            'weight' => 1,
+                        ],
+                    ],
+                ],
+                [
+                    'name' => 'child',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'dependencies' => [
+                        [
+                            'feature' => 'parent',
+                            'variants' => ['variant1'],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $this->pushResponse($data);
+        $instance = $this->getCachedInstance();
+        self::assertTrue($instance->isEnabled('child'));
+    }
+
+    public function testDependenciesMixedResolvedUnresolved()
```

Comment:
This test might be worth adding to client specs.

Reply:
Can you confirm if "Child is enabled if its parent matches any of the specified variants." is covering this case?